kafka-impl/src/main/java/io/streamnative/pulsar/handlers/kop/security/auth/SimpleAclAuthorizer.java

```diff
@@ -0,0 +1,147 @@
+            if (exception != null) {
+                log.error("Check super user error: {}", exception.getMessage());
+                return;
+            }
+            if (isSuper) {
+                permissionFuture.complete(true);
+            } else {
+                try {
+                    getPulsarService()
+                            .getPulsarResources()
+                            .getNamespaceResources()
+                            .getAsync(String.format("%s/%s", POLICY_ROOT, topicName.getNamespace()))
+                            .thenAccept(policies -> {
+                                if (!policies.isPresent()) {
+                                    if (log.isDebugEnabled()) {
+                                        log.debug("Policies node couldn't be found for namespace : {}", principal);
+                                    }
+                                } else {
+                                    String role = principal.getName();
+                                    Map<String, Set<AuthAction>> topicRoles = policies.get()
+                                            .auth_policies
+                                            .getTopicAuthentication()
+                                            .get(topicName.toString());
+                                    if (topicRoles != null && role != null) {
+                                        // Topic has custom policy
+                                        Set<AuthAction> topicActions = topicRoles.get(role);
+                                        if (topicActions != null && topicActions.contains(action)) {
+                                            permissionFuture.complete(true);
+                                            return;
+                                        }
+                                    }
+
+                                    Map<String, Set<AuthAction>> namespaceRoles = policies.get().auth_policies
+                                            .getNamespaceAuthentication();
+                                    Set<AuthAction> namespaceActions = namespaceRoles.get(role);
+                                    if (namespaceActions != null && namespaceActions.contains(action)) {
+                                        permissionFuture.complete(true);
+                                        return;
+                                    }
+                                }
+                                permissionFuture.complete(false);
+                            }).exceptionally(ex -> {
+                                log.warn("Client with Principal - {} failed to get permissions for resource - {}. {}",
+                                        principal, topicName, ex.getMessage());
+                                permissionFuture.completeExceptionally(ex);
+                                return null;
+                            });
+                } catch (Exception e) {
```

Comment:
Don't catch `Exception`, catch specific exception types, which are usually checked exceptions, but sometimes catching unchecked exceptions is also acceptable.